Common/Data/Auxiliary/FactorFile.cs

```diff
@@ -208,26 +220,137 @@ public bool HasSplitEventOnNextTradingDay(DateTime date, out decimal splitFactor
             return false;
         }
 
+        /// <summary>
+        /// Writes this factor file data to an enumerable of csv lines
+        /// </summary>
+        /// <returns>An enumerable of lines representing this factor file</returns>
+        public IEnumerable<string> ToCsvLines()
+        {
+            if (FactorFileMinimumDate != null)
+            {
+                var min = SortedFactorFileData.First().Value;
+                yield return $"{FactorFileMinimumDate:yyyyMMdd},{min.PriceFactor},{min.SplitFactor}";
+            }
+
+            foreach (var kvp in SortedFactorFileData.OrderBy(kvp => kvp.Value.Date))
+            {
+                yield return $"{kvp.Key:yyyyMMdd},{kvp.Value.PriceFactor},{kvp.Value.SplitFactor}";
+            }
+        }
+
         /// <summary>
         /// Write the factor file to the correct place in the default Data folder
         /// </summary>
         /// <param name="symbol">The symbol this factor file represents</param>
         public void WriteToCsv(Symbol symbol)
         {
             var filePath = LeanData.GenerateRelativeFactorFilePath(symbol);
+            File.WriteAllLines(filePath, ToCsvLines());
+        }
 
-            var csv = new StringBuilder();
+        /// <summary>
+        /// Gets all of the splits and dividends represented by this factor file
+        /// </summary>
+        /// <param name="symbol">The symbol to ues for the dividend and split objects</param>
+        /// <param name="exchangeHours">Exchange hours used for resolving the previous trading day</param>
+        /// <returns>All splits and diviends represented by this factor file in chronological order</returns>
+        public List<BaseData> GetSplitsAndDividends(Symbol symbol, SecurityExchangeHours exchangeHours)
+        {
+            var dividendsAndSplits = new List<BaseData>();
+            var futureFactorFileRow = SortedFactorFileData.Last().Value;
+            for (var i = SortedFactorFileData.Count - 2; i >= 0 ; i--)
```

Comment:
Not sure if I'm right, but if using `Count - 1` shouldn't you be able to remove lines `260` and `276`?

Reply:
what would I pass to the `GetDividend` method if I haven't initialized `futureFactorFileRow` first? I figured doing it this way removed an `if` statement from the loop detecting the first iteration which has the year 2050 entry.

Reply:
Your right! I analyzed the code wrong 👍